src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,350 @@
+                }
+
+            } catch (DukeException e) {
+                Ui.customErrorMessage(e.getMessage());
+            }
+        } else if (input.equals("list")) {
+            response = Duke.printList();
+        } else if (input.equals("bye")) {
+            Duke.makeInactive();
+            response = Ui.sendEndMessage();
+            Storage.rewriteFile(Duke.getTaskList());
+        } else if (input.split(" ")[0].equals("delete")) {
+
+            try {
+                if (input.split(" ").length == 1) {
+                    throw new DukeException(
+                            "____________________________________________________________\n"
+                                    + "☹ OOPS!!! Please state the task number that you want to delete.\n"
+                                    + "____________________________________________________________"
+                    );
+                }
+                String taskToBeDeleted = input.split(" ")[1];
+                response = Duke.deleteTask(taskToBeDeleted);
+
+                //rewrite whole txt file
+                Storage.rewriteFile(Duke.getTaskList());
+            } catch (DukeException e) {
+                Ui.customErrorMessage(e.getMessage());
+            }
+
+        } else if (input.split(" ")[0].equals("find")) {
+            try {
+                if (input.split(" ").length == 1) {
+                    throw new DukeException(
+                            "____________________________________________________________\n"
+                                    + "☹ OOPS!!! Please tell me what you want to find!\n"
+                                    + "____________________________________________________________"
+                    );
+                }
+
+                String keyword = input.split(" ")[1];
+                TaskList matchingTasks = Duke.findTask(keyword);
+
+                if (matchingTasks.size() == 0) {
+                    response = "____________________________________________________________\n"
+                            + "☹ OOPS!!! No tasks match your search.\n"
+                            + "____________________________________________________________";
+                } else {
+                    int number = 1;
+                    response = "____________________________________________________________\n"
+                            + "Here is what I found:\n";
+                    for (int i = 0; i < matchingTasks.size(); i++) {
+                        //System.out.println(number + "." + matchingTasks.get(i).toString());
+                        response += number + "." + matchingTasks.get(i).toString() + "\n";
+                        number++;
+                    }
+                }
+            } catch (DukeException e) {
+                Ui.customErrorMessage(e.getMessage());
+            }
+        } else {
+            response = "____________________________________________________________\n"
```

Comment:
Since there are multiple instances to draw the lines for formatting, you could consider having a method to print the line to make your code cleaner!